app/src/main/java/fr/free/nrw/commons/nearby/NearbyActivity.java

```diff
@@ -495,4 +505,25 @@ public void onLocationChangedSlightly(LatLng latLng) {
     public void prepareViewsForSheetPosition(int bottomSheetState) {
         // TODO
     }
+    private boolean haveNetworkConnection() {
+
```

Comment:
Please move this method to a new class named NetworkUtils as a public static method. So that is can be used on every network check